apps/tailwind-components/components/input/Tree.vue

```diff
@@ -51,11 +49,15 @@ function expandSelection(node: ITreeNode) {
 
 function toggleSelect(node: ITreeNode) {
   if (props.modelValue.includes(node.name)) {
-    // remove node from selected nodes and emit new model
-    emit(
-      "update:modelValue",
-      props.modelValue.filter((n) => n !== node.name)
+    // remove node(s) from selected nodes and emit new model
+    let deSelectionList = [node.name];
+    if (props.expandSelected) {
+      deSelectionList = deSelectionList.concat(expandSelection(node));
```

Comment:
expand selection return a expansion list , it can bew used for either selection of deselection